corehq/apps/reports/standard/deployments.py

```diff
@@ -82,58 +81,56 @@ def selected_app_id(self):
 
     @property
     @memoized
-    def users(self):
+    def user_query(self):
         mobile_user_and_group_slugs = self.request.GET.getlist(ExpandedMobileWorkerFilter.slug)
 
-        limit_user_ids = []
-        if self.selected_app_id:
-            limit_user_ids = get_all_user_ids_submitted(self.domain, self.selected_app_id)
-
-        users_data = ExpandedMobileWorkerFilter.pull_users_and_groups(
+        user_query = ExpandedMobileWorkerFilter.user_es_query(
             self.domain,
             mobile_user_and_group_slugs,
-            include_inactive=False,
-            limit_user_ids=limit_user_ids,
         )
-        return users_data.combined_users
+        user_query = (user_query
+                      .set_sorting_block(self.get_sorting_block())
+                      .size(self.pagination.count)
+                      .start(self.pagination.start))
+        if self.selected_app_id:
+            user_query = user_query.filter(
+                filters.term('reporting_metadata.last_submissions.app_id', self.selected_app_id)
+            )
+        return user_query.run()
+
+    @property
+    def total_records(self):
+        return self.user_query.total
 
     @property
     def rows(self):
-        rows = []
 
-        user_ids = map(lambda user: user.user_id, self.users)
-        user_xform_dicts_map = get_last_form_submissions_by_user(self.domain, user_ids, self.selected_app_id)
+        rows = []
 
-        for user in self.users:
-            xform_dict = last_seen = last_sync = app_name = None
+        for user in self.user_query.hits:
+            last_seen = last_sync = app_name = None
             app_version_info_from_form = app_version_info_from_sync = None
-            if user_xform_dicts_map.get(user.user_id):
-                xform_dict = user_xform_dicts_map[user.user_id][0]
+            last_submissions = user.get('reporting_metadata', {}).get('last_submissions')
 
-            if xform_dict:
-                last_seen = string_to_utc_datetime(xform_dict.get('received_on'))
+            if last_submissions:
+                last_sub = sorted(last_submissions, key=lambda sub: sub.get('submission_date'), reverse=True)[0]
+                last_seen = string_to_utc_datetime(last_sub.get('submission_date'))
 
-                if xform_dict.get('app_id'):
+                if last_sub.get('app_id'):
                     try:
-                        app = get_app(self.domain, xform_dict.get('app_id'))
+                        app = get_app(self.domain, last_sub['app_id'])
                     except ResourceNotFound:
                         pass
                     else:
                         app_name = app.name
```

Comment:
If all we need is the app name then a well cached util function would be good. Either way the app lookup should be heavily cached.